src/OptFrame/Evaluation.hpp

```diff
@@ -67,6 +67,7 @@ typedef EVALUATION_TYPE evtype;
 // note: for multi-objective problems with distinct objective space types
 // such as (int, evtype, long long) you can use PackTypes in Utils or overload
 // manually each of the numeric operators +, -, *
+//why this isn't a template????????????????
```

Comment:
It's complicated, sorry it's not a template, but it's really problematic (I don't know how to solve).

On Multi Objective problems, you require two or more simulataneous Evaluators... and since they all return Evaluation type, you cannot  template it (Evaluation<type>), and at the same time, create a vector out of it (see MultiEvaluation class).
The only possible solution would be to replace all MultiEvaluation internal vector by a std::array, of specific types for each different evaluation function... some objectives may be integer, others are double, ..., so we decided for double, since it could embed integers as well.
On general, it's either double or integer (perhaps BigInteger, decimal, or BigDecimal) we need to see how we handle this.